File: niuniu-take-out-springboot3/server/src/main/java/common/niuniu/service/ReportService.java
package common.niuniu.service;

import common.niuniu.vo.OrderReportVO;
import common.niuniu.vo.SalesTop10ReportVO;
import common.niuniu.vo.TurnoverReportVO;
import common.niuniu.vo.UserReportVO;

import jakarta.servlet.http.HttpServletResponse;
import java.time.LocalDate;

public interface ReportService {
    TurnoverReportVO getTurnover(LocalDate begin, LocalDate end);

    UserReportVO getUser(LocalDate begin, LocalDate end);

    OrderReportVO getOrder(LocalDate begin, LocalDate end);

    SalesTop10ReportVO getSalesTop10(LocalDate begin, LocalDate end);

    void exportBusinessData(HttpServletResponse response);
}

File: niuniu-take-out-kotlin/server/src/main/kotlin/common/niuniu/service/ReportService.kt
package common.niuniu.service

import common.niuniu.vo.OrderReportVO
import common.niuniu.vo.SalesTop10ReportVO
import common.niuniu.vo.TurnoverReportVO
import common.niuniu.vo.UserReportVO
import jakarta.servlet.http.HttpServletResponse
import java.time.LocalDate

interface ReportService {
    fun getTurnover(begin: LocalDate?, end: LocalDate?): TurnoverReportVO?

    fun getUser(begin: LocalDate?, end: LocalDate?): UserReportVO?

    fun getOrder(begin: LocalDate?, end: LocalDate?): OrderReportVO?

    fun getSalesTop10(begin: LocalDate?, end: LocalDate?): SalesTop10ReportVO?

    fun exportBusinessData(response: HttpServletResponse?)
}
